back-end/bal-forum/tests/bal_forum_service_test.bal

```diff
@@ -0,0 +1,390 @@
+import ballerina/http;
+import ballerina/sql;
+import ballerina/test;
+import ballerina/time;
+import ballerina/uuid;
+import ballerinax/mysql;
+import ballerinax/nats;
+
+configurable string forumServiceUrl = "http://localhost:4000/api";
+
+http:Client forumClient = check new (forumServiceUrl);
+
+public type Post record {
+    string text;
+};
+
+client class MockHttpClient {
+    isolated resource function post [http:PathParamType... path](http:RequestMessage message, map<string|string[]>? headers = (), string?
```

Comment:
Can't we mock the exact resource function here rather than having default one?

Reply:
As of now, we do not have a way to mock resource functions. The workaround for this is creating a test double. Hence this is the only way to mock the sentiment client.